gm_dotnet_managed/GmodNET/GameConsoleWriter.cs

```diff
@@ -0,0 +1,45 @@
+using System;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Text;
+
+public class GameConsoleWriter : TextWriter
+{
+    private class Tier0
+    {
+        [DllImport("tier0", CallingConvention = CallingConvention.Cdecl)]
+        public static extern void Msg([MarshalAs(UnmanagedType.LPStr)] string msg);
+    }
+
+    private static GameConsoleWriter instance;
+    private uint references;
+
+    public override string NewLine { get => "\n"; }
+    public override Encoding Encoding => throw new NotImplementedException();
+
+    public override void Write(string value)
+    {
+        Tier0.Msg(value);
+    }
+
+    public static void Load()
+    {
+        if (instance == null)
+        {
+            instance = new GameConsoleWriter();
+            Console.SetOut(instance);
+        }
+        instance.references++;
+    }
+    public static void Unload()
+    {
+        if (instance != null)
+        {
+            instance.references--;
+            if (instance.references <= 0)
+            {
+                instance = null;
+            }
+        }
+    }
```

Comment:
I do not understand why do we need to load and unload console writer. It seem like you are confused with a lifetime of this method, aren't you?

Reply:
Yes

Reply:
@SupinePandora43 It seems like you shouldn't load and unload `Console.SetOut` per `GlobalContext`. Try to make call `Cosnole.SetOut` just once in `Startup.Main`. It should work.

Reply:
I'll do it tomorrow.
Instead of `GameConsoleWriter.Load();`
Use `Console.SetOut(new GameConsoleWriter());`?

Reply:
Yes, give it a try!

Reply:
resolved in https://github.com/GmodNET/GmodDotNet/pull/70/commits/d2f865e987f0302303e76a918edc08efcb55f5ae